crud/readview.lua

```diff
@@ -286,6 +286,15 @@ end
 function Readview_obj.create(vshard_router, opts)
     local readview = {}
     setmetatable(readview, Readview_obj)
+
+    -- For tarantool lua (and lua 5.1) __gc metamethod only works for cdata types.
+    -- So in order to create a proper GC hook, we need to create cdata with
+    -- __gc call.
+    -- __gc call for this cdata will be a __gc call for our readview.
+    -- https://github.com/tarantool/tarantool/issues/5770
+    local proxy = newproxy(true)
```

Comment:
It doesn't seems right. `proxy` will have zero references after `create`, so its `__gc` may trigger before readbiew object `__gc`. For example, you may borrow existing hack from here

https://github.com/tarantool/crud/blob/416b88018933ee68961268163d8bf92922652b1b/crud/stats/init.lua#L260-L268

Reply:
changed, added test for that case